daemon/protocol/client_lib.py

```diff
@@ -0,0 +1,228 @@
+    _client = daemon_pb2.beta_create_ScootDaemon_stub(channel)
+
+
+
+def stop():
+    """
+    """
+    global _client
+    if not is_started():
+        raise ScootException("Already stopped.")
+    _client = None
+    pass
+
+
+
+def is_started():
+    """
+    """
+    global _client
+    return _client is not None
+
+
+def echo(ping):
+    """
+    """
+    global _client
+    if not is_started():
+        raise ScootException("Not started.")
+    req = daemon_pb2.EchoRequest(ping=ping)
+    try: resp = _client.Echo(req)
+    except Exception as e: raise ScootException(e)
+    return resp.pong
+
+
+def create_snapshot(path):
+    """
+    """
+    global _client, _context
+    if not is_started():
+        raise ScootException("Not started.")
+    req = daemon_pb2.CreateSnapshotRequest(path=path)
+    try: resp = _client.CreateSnapshot(req, _context)
+    except Exception as e: raise #ScootException(e)
+    if resp.error != "":
+        raise ScootException(resp.error)
+    return resp.snapshot_id
+
+
+def checkout_snapshot(snapshot_id, dirpath):
+    """
+    """
+    global _client
+    if not is_started():
+        raise ScootExceptoin("Not started.")
+    req = daemon_pb2.CheckoutSnapshotRequest(snapshot_id=snapshot_id, dir=dirpath)
+    try: resp = _client.CheckoutSnapshot(req)
+    except Exception as e: raise ScootException(e)
+    if resp.error != "":
+        raise ScootException(resp.error)
+    return
+
+
+def run(snapshot_id, argv, env=None, timeout_ns=0, src_paths_to_dest_dirs=None, ):
+    """
+    """
+    global _client
+    if not is_started():
+        raise ScootExceptoin("Not started.")
+    cmd = daemon_pb2.RunRequest.Command(timeout_ns=timeout_ns)
+    cmd.argv.extend(argv)
+    for key, val in env.iteritems():
+        cmd.env[key] = val
+
+    plan = daemon_pb2.RunRequest.OutputPlan()
+    for key, val in src_paths_to_dest_dirs.iteritems():
+        plan.src_paths_to_dest_dirs[key] = val
+
+    req = daemon_pb2.RunRequest(snapshot_id=snapshot_id, cmd=cmd, plan=plan)
+    try: resp = _client.Run(req)
+    except Exception as e: raise ScootException(e)
+    if resp.error != "":
+        raise ScootException(resp.error)
+    return resp.run_id
+
+
+def poll(run_ids, timeout_ns=0, return_all=False):
+    """
+    """
+    global _client
+    if not is_started():
+        raise ScootExceptoin("Not started.")
+    req = daemon_pb2.PollRequest(timeout_ns=timeout_ns, all=return_all)
+    req.run_ids.extend(run_ids)
+    try: resp = _client.Poll(req)
+    except Exception as e: raise ScootException(e)
+
+    def domain(status):
+        return ScootStatus(run_id=status.run_id,
+                           state=status.state,
+                           snapshot_id=status.snapshot_id,
+                           exit_code=status.exit_code,
+                           error=status.error)
+    return [domain(x) for x in resp.status]
+
+
+if __name__ == '__main__':
```

Comment:
at a minimum, it'd be good to move this into a function defn (which at least nets a function scope vs global) and then do:

```
if __name__ == '__main__':
  run_test()
```